packages/trader/src/Modules/Trading/Containers/trade-params-mobile.jsx

```diff
@@ -20,135 +20,133 @@ const DEFAULT_DURATION = Object.freeze({
 const makeGetDefaultDuration = (trade_duration, trade_duration_unit) => duration_unit =>
     trade_duration_unit === duration_unit ? trade_duration : DEFAULT_DURATION[duration_unit];
 
-class TradeParamsModal extends React.Component {
-    constructor(props) {
-        super(props);
-        const { amount, duration, duration_unit } = this.props;
-        const getDefaultDuration = makeGetDefaultDuration(duration, duration_unit);
+const TradeParamsModal = props => {
+    const { amount, duration, duration_unit } = props;
 
-        this.state = {
-            trade_param_tab_idx: 0,
-            duration_tab_idx: undefined,
-            amount_tab_idx: undefined,
-            has_amount_error: false,
-            has_duration_error: false,
-            // duration unit values
-            curr_duration_unit: duration_unit,
-            curr_duration_value: duration,
-            t_duration: getDefaultDuration('t'),
-            s_duration: getDefaultDuration('s'),
-            m_duration: getDefaultDuration('m'),
-            h_duration: getDefaultDuration('h'),
-            d_duration: getDefaultDuration('d'),
-            // amount values
-            stake_value: amount,
-            payout_value: amount,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const getDefaultDuration = React.useCallback(makeGetDefaultDuration(duration, duration_unit), []);
+
+    const [state, setState] = React.useState({
+        trade_param_tab_idx: 0,
+        duration_tab_idx: undefined,
+        amount_tab_idx: undefined,
+        has_amount_error: false,
+        has_duration_error: false,
+        // duration unit values
+        curr_duration_unit: duration_unit,
+        curr_duration_value: duration,
+        t_duration: getDefaultDuration('t'),
+        s_duration: getDefaultDuration('s'),
+        m_duration: getDefaultDuration('m'),
+        h_duration: getDefaultDuration('h'),
+        d_duration: getDefaultDuration('d'),
+        // amount values
+        stake_value: amount,
+        payout_value: amount,
+    });
+
+    React.useEffect(() => {
+        // Fix to prevent iOS from zooming in erratically on quick taps
+        const preventIOSZoom = event => {
+            if (event.touches.length > 1) {
+                event.preventDefault();
+                event.stopPropagation();
+            }
```

Comment:
Does this happens elsewhere? I mean we can refactor this function to `shared` package?

Reply:
moved `preventIOSZoom` hook to shared